app/main.py

```diff
@@ -1,3 +1,31 @@
-def cache(func):
-    # Write your code here
-    pass
+from functools import wraps
+from typing import Callable, ParamSpec, TypeVar
+
+Param = ParamSpec("Param")
+RetType = TypeVar("RetType")
+
+
+def cache(func: Callable[Param, RetType]) -> Callable[Param, RetType]:
+    cache_dict: dict = {}
+
+    @wraps(func)
+    def wrapper(*args, **kwargs) -> RetType:
+        for arg in args:
+            try:
+                hash(arg)
+            except TypeError:
+                return func(*args, **kwargs)
+        for value in kwargs.values():
+            try:
+                hash(value)
+            except TypeError:
+                return func(*args, **kwargs)
+        signature: tuple = (args, tuple(kwargs.items()))
```

Comment:
We're caching all the values from args and kwargs as a key?

Reply:
i thought that using args and kwargs as a key would prevent returning the wrong output if there is a keyword argument that changes the function's return value